ebpf/port.go

```diff
@@ -0,0 +1,75 @@
+package ebpf
+
+import (
+	"fmt"
+	log "github.com/cihub/seelog"
+	"path"
+	"time"
+)
+
+// PortMapping tracks which ports a pid is listening on
+type PortMapping struct {
+	procRoot string
+	config   *Config
+	ports    map[uint16]string
+}
+
+//NewPortMapping creates a new PortMapping instance
+func NewPortMapping(procRoot string, config *Config) *PortMapping {
+	return &PortMapping{
+		procRoot: procRoot,
+		config:   config,
+		ports:    make(map[uint16]string),
+	}
+}
+
+// AddMapping indicates that something is listening on the provided address and port
+func (pm *PortMapping) AddMapping(port uint16, address string) {
+	pm.ports[port] = address
+}
+
+// RemoveMapping indicates that the provided port is no longer being listened on
+func (pm *PortMapping) RemoveMapping(port uint16) {
+	delete(pm.ports, port)
+}
+
+// IsListening returns true if something is listening on the given address and port
+func (pm *PortMapping) IsListening(port uint16, address string) bool {
+	listenAddr, ok := pm.ports[port]
+	if !ok {
+		return false
+	}
+
+	return listenAddr == address
+}
+
+// ReadInitialState reads the /proc filesystem and determines which pids are currently listening on what ports
+func (pm *PortMapping) ReadInitialState() error {
+	start := time.Now()
+
+	if pm.config.CollectTCPConns {
+		ports, err := readProcNet(path.Join(pm.procRoot, "net/tcp"))
+		if err != nil {
+			return fmt.Errorf("error reading tcp state: %s", err)
```

Comment:
> We might consider a softer failure where we log it but move on, since direction isn't strictly required (is it?) 
No, I don't think so

> and it would get picked up anyway for newer connections w/ bpf events.
Correct.  We would only miss out on identifying tcp connections that were established before the tracer started.